src/daemon.rs

```diff
@@ -58,34 +58,82 @@ pub fn main() {
     for hotkey in &hotkeys {
         log::debug!("hotkey: {:#?}", hotkey);
     }
-
-    match sock_send("notify-send hello world") {
-        // testing .
-        Err(e) => {
-            log::error!("Failed to send command over IPC.");
-            log::error!("Is swhks running?");
-            log::error!("{:#?}", e)
-        }
-        _ => {}
+    if let Err(e) = sock_send("notify-send hello world") {
+        log::error!("Failed to send command over IPC.");
+        log::error!("Is swhks running?");
+        log::error!("{:#?}", e)
     };
 
+    // Super,
+    // Hyper,
+    // Meta,
+    // Alt,
+    // Control,
+    // Shift,
+    // ModeSwitch,
+    // Lock,
+    // Mod1,
+    // Mod2,
+    // Mod3,
+    // Mod4,
+    // Mod5,
+
+    //a hashmap that map evdev:Key to enum modifiers
+    let modifiers_map: HashMap<Key, config::Modifier> = HashMap::from([
+        (Key::KEY_LEFTMETA, config::Modifier::Super),
+        (Key::KEY_RIGHTMETA, config::Modifier::Super),
+        (Key::KEY_LEFTMETA, config::Modifier::Super),
+        (Key::KEY_RIGHTMETA, config::Modifier::Super),
+        (Key::KEY_LEFTALT, config::Modifier::Alt),
+        (Key::KEY_RIGHTALT, config::Modifier::Alt),
+        (Key::KEY_LEFTCTRL, config::Modifier::Control),
+        (Key::KEY_RIGHTCTRL, config::Modifier::Control),
+        (Key::KEY_LEFTSHIFT, config::Modifier::Shift),
+        (Key::KEY_RIGHTSHIFT, config::Modifier::Shift),
+    ]);
+
     let mut key_states: Vec<AttributeSet<Key>> = Vec::new();
     let mut possible_hotkeys: Vec<config::Hotkey> = Vec::new();
     loop {
         for device in &keyboard_devices {
             key_states.push(device.get_key_state().unwrap());
         }
-
+        // check if a hotkey in hotkeys is pressed
         for state in &key_states {
             for hotkey in &hotkeys {
-                if state.iter().count() == hotkey.modifiers.len() + 1 {
-                    // +1 Because we handle only 1 keysym.
+                if hotkey.modifiers.len() < state.iter().count() {
```

Comment:
why not check the exact count?

Reply:
Because if we have multiple keysyms pressed, we'll still run the command once the correct keysym is pressed.
`if state_modifiers == hotkey.modifiers && state_keysyms.contains(&hotkey.keysym)`
And that's also why I use `.contains` here.

Reply:
> Because if we have multiple keysyms pressed, we'll still run the command once the correct keysym is pressed. `if state_modifiers == hotkey.modifiers && state_keysyms.contains(&hotkey.keysym)` And that's also why I use `.contains` here.

Does sxhkd do that? The last time I tested, it didn't.

Reply:
![image](https://user-images.githubusercontent.com/60808802/152916394-da0d64ce-5a71-4078-a9d7-661357778085.png)

Reply:
> 

Here when I press `super + shift + a` and then `w` while still holding `super` , `shift` and `a`, firefox (which is supposed to open when `super + shift + w` pressed) opens.

Reply:
@Idlidev tested it and confirmed that with multiple non modifier keys, it does work but not with multiple modifier keys, has this been taken into account?

Reply:
> > 
> 
> Here when I press `super + shift + a` and then `w` while still holding `super` , `shift` and `a`, firefox (which is supposed to open when `super + shift + w` pressed) opens.

Alright, got it